File: spark-project/web/src/main/java/com/aura/database/DBHelper.java
package com.aura.database;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class DBHelper {

	/**
	 * 事务提交
	 */
	public static void commit(Connection conn){
		try {
			if (conn != null){
				conn.commit();
				System.out.println("do commit...");
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	/**
	 * 事务处理
	 */
	public static void setAutoCommit(Connection conn, boolean autoCommit){
		try {
			if (conn != null)
				conn.setAutoCommit(autoCommit);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	/**
	 * 批处理提交
	 */
	public static void executeBatch(Connection conn, PreparedStatement pstmt, int count) throws SQLException {
		if(count % 1000 == 0) {
			pstmt.executeBatch();
			DBHelper.commit(conn);
		} 
	}
	
	/**
	 * 释放rs stmt conn
	 */
	public static void close(Connection conn, PreparedStatement pstmt, ResultSet rs) {
		close(rs);
		close(pstmt);
		close(conn);
	}
	
	/**
	 * 释放conn
	 */
	public static void close(Connection conn){
		try {
			if (conn != null)
				conn.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	/**
	 * 释放pstmt
	 */
	public static void close(PreparedStatement pstmt){
		try {
			if (pstmt != null)
				pstmt.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	/**
	 * 释放rs
	 */
	public static void close(ResultSet rs){
		try {
			if (rs != null)
				rs.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	public static void main(String[] args) throws Exception {
		
	}

}

File: spark-project/analysis/src/main/scala/com/aura/db/DBHelper.scala
package com.aura.db

import java.sql.{Connection, PreparedStatement, ResultSet}

object DBHelper {
  
  /**
   * 获得默认的数据库连接
   */
  def getConnection(): Connection = {
    // return JDBCHelper.getConnection()
		C3P0Helper.getConnection()
	}
  /**
	 * 事务提交
	 */
	def commit(conn: Connection): Unit = {
	  if (conn != null){
	    conn.commit()
			println("do commit...")
	  }
	}
	
	/**
	 * 事务处理
	 */
	def setAutoCommit(conn: Connection, autoCommit: Boolean): Unit = {
	  if (conn != null) {
	    conn.setAutoCommit(autoCommit)
	  }
	}
	
	/**
	 * 批处理提交
	 */
	def executeBatch(conn: Connection, pstmt: PreparedStatement, count: Int): Unit = {
	  if(count % 1000 == 0) {
	    pstmt.executeBatch()
	    DBHelper.commit(conn)
  	} 
	}
	
	/**
	 * 释放rs stmt conn
	 */
	def close(conn: Connection, pstmt: PreparedStatement, rs: ResultSet): Unit = {
	  close(rs)
		close(pstmt)
		close(conn)
	}
	
	/**
	 * 释放conn
	 */
	def close(conn: Connection): Unit = {
	  if (conn != null) {
	    conn.close()
	  }
	}
	
	/**
	 * 释放pstmt
	 */
	def close(pstmt: PreparedStatement): Unit = {
	  if (pstmt != null) {
	    pstmt.close()
	  }
	}
	
	/**
	 * 释放rs
	 */
	def close(rs: ResultSet): Unit = {
	  if (rs != null) {
	    rs.close()
	  }
	}
  
  def main(args: Array[String]): Unit = {
    
  }
}